py/picca/io.py

```diff
@@ -232,16 +238,60 @@ def read_blinding(in_dir):
                                                            + '/*.fits.gz')
     filename = files[0]
     hdul = fitsio.FITS(filename)
-    if "LAMBDA" in hdul: # This is for ImageHDU format
-        header = hdul["METADATA"].read_header()
-        blinding = header["BLINDING"]
-    else: # This is for BinTable format
+    # This is for ImageHDU format
+    if "LAMBDA" in hdul:
+        header_blinding = hdul["METADATA"].read_header()
+        header_lambda = hdul["LAMBDA"].read_header()
+
+        # read blinding
+        blinding = header_blinding["BLINDING"]
+
+        # check if we need to blind
+        if tracer2 is None:
+            tracer2 = tracer1
+        blinding_tracers = ["LYA", "QSO"]
+        if tracer1 not in blinding_tracers or tracer2 not in blinding_tracers:
+            # one of the tracers is not relevant, so do not blind (ever)
+            blinding = "none"
+        else:
+            # check if we are doing metal forests
+            if "MIN_RF_WAVE" in header_lambda and "MAX_RF_WAVE" in header_lambda:
+                lya_wave = ABSORBER_IGM["LYA"]
+                lyb_wave = ABSORBER_IGM["LYB"]
+                min_rf_wave = header_lambda["MIN_RF_WAVE"]
+                max_rf_wave = header_lambda["MAX_RF_WAVE"]
+                if ((min_rf_wave > lya_wave and max_rf_wave > lya_wave) or
+                    (min_rf_wave < lyb_wave and max_rf_wave < lyb_wave)):
+                    # we are doing a metal forest, so do not blind (ever)
+                    blinding = "none"
```

Comment:
@iprafols , @Andrea-MG - Are you sure about this? It looks to me that the this code would not blind the QSO x LyA(LyB) correlation. 

I believe the "if" statement in line 263 should only have the first line (check that both rf_wave are above lya_wave), but it should not have the second condition.